drivers/tcu_pynq/driver.py

```diff
@@ -211,6 +213,50 @@ def load_model(self, model_filename):
         with open(d + self.model.prog.file_name, "rb") as f:
             self.program = f.read()
 
+    def scalar_address(self, vec_address):
+        return vec_address * self.arch.array_size
+
+    def prepare_flush_probe(self): 
+        # initialize flush probe
+        self.probe_source_address = self.arch.dram0_depth - 1
+        self.probe_target_address = self.arch.dram0_depth - 2
+        self.probe_source = np.full(
+            self.arch.array_size,
+            np.iinfo(data_type_numpy(self.arch.data_type)).max,
+            dtype=data_type_numpy(self.arch.data_type))
+        self.probe_target = np.full(
+            self.arch.array_size,
+            0,
+            dtype=data_type_numpy(self.arch.data_type))
+        
+        # write flush probe
+        self.dram0.write(
+            self.scalar_address(self.probe_source_address),
+            self.probe_source)
+        self.dram0.write(
+            self.scalar_address(self.probe_target_address),
+            self.probe_target)
+
+        # flush probe instructions
+        return [
+            self.layout.data_move(
+                DataMoveFlag.dram0_to_memory,
+                0,
+                self.probe_source_address,
+                0),
+            self.layout.data_move(
+                DataMoveFlag.memory_to_dram0,
+                0,
+                self.probe_target_address,
+                0)
+        ]
+
+    def wait_for_flush(self):
+        while not self.dram0.compare(
+            self.scalar_address(self.probe_target_address),
+            self.probe_source):
+            pass
```

Comment:
Might it make more sense to do something like `time.sleep(0.001)` here?

Reply:
I was experimenting with small sleeps instead of flushing and on Ultra96 it seems like even smallest sleeps were sufficient for data to arrive, which means true delay is even smaller. With this adding small sleep in the loop may actually add unnecessary latency.